### ISSUE TITLE
 AGP Pruner may be messed by the bias of batch normalization?

### ISSUE THREAD

**Issue by twmht** [microsoft/nni#3848]
training from model_speedup() is good since the bias of batch normalization can be masked out.

The [example](https://github.com/microsoft/nni/blob/master/examples/model_compress/end2end_compression.py)  shows an end to end trainer, where the model is speed up before fine tuning.

However, Some iterative pruner like AGP pruner, which would update the mask after some iterations. So it can't speed up before fine tuning.  But the zero-out weight would be active by the bias of batch normalization during traning.  that is, the output of `apply_compression_result` is not same with `model_speedup()`.

is this a serious problem? In small problem like cifar10 or mnist, this might won't be a problem, but what about some large scale training data, like imagenet?
